cellprofiler_core/utilities/image.py

```diff
@@ -301,16 +304,37 @@ def download_to_temp_file(url):
             Params={'Bucket': bucket_name, 'Key': key.replace("+", " ")}
         )
 
-    from urllib.request import urlopen
-    logging.info(f"Downloading image from {url}")
-    src = urlopen(url)
-    dest_file = tempfile.NamedTemporaryFile(suffix=ext, delete=False, dir=CP_TEMP_DIR.name)
-    try:
-        shutil.copyfileobj(src, dest_file)
-    except Exception as e:
-        logging.error(f"Unable to download image to temp file. {e}")
-        return None
-    finally:
-        dest_file.close()
-    return dest_file.name
+    elif scheme == 'gs':
+            # Get default Google Cloud Storage project ID from environment.
+            if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
```

Comment:
unnecessary line of code?

Reply:
Do you mean that the line `if `GOOGLE_APPLICATION_CREDENTIALS` in os.environ`? If I were to throw an exception when that condition is false, would that enable me to keep that line?

Reply:
No need to throw an exception! The GCS Python client will throw an exception if it doesn't have the necessary auth.

Reply:
Okay, thank you! I have removed the unnecessary lines.